### PR TITLE
Add missing doc strings

### PR DESCRIPTION
**Pull request by gilbertw**
Fixes a some of the places where doc strings are missing.  Does not address all the platform headers

### CODE REVIEW

File: sdk/platform/http_client/nohttp/src/az_nohttp.c [hunk 1]
```diff
@@ -5,6 +5,14 @@
 
 #include <_az_cfg.h>
 
+/**
+ * @brief Provides no http support.
```
Comment (ahsonkhan) [inline sdk/platform/http_client/nohttp/src/az_nohttp.c:9]:
```suggestion
 * @brief Provides no HTTP support.
```

File: sdk/platform/http_client/nohttp/src/az_nohttp.c [hunk 2]
```diff
@@ -5,6 +5,14 @@
...
+ *
+ * @param p_request an internal http builder with data to build and send http request
```
Comment (ahsonkhan) [inline sdk/platform/http_client/nohttp/src/az_nohttp.c:11]:
```suggestion
 * @param p_request An internal HTTP builder with data to build and send HTTP request.
```

File: sdk/platform/http_client/nohttp/src/az_nohttp.c [hunk 3]
```diff
@@ -5,6 +5,14 @@
...
+ * @param p_response pre-allocated buffer where http response will be written
```
Comment (ahsonkhan) [inline sdk/platform/http_client/nohttp/src/az_nohttp.c:12]:
```suggestion
 * @param p_response A pre-allocated buffer where the HTTP response will be written.
```

File: sdk/core/core/inc/az_credentials.h [hunk 4]
```diff
@@ -104,8 +104,9 @@ typedef struct
  * @param tenant_id an Azure tenant ID
  * @param client_id an Azure client ID
  * @param client_secret an Azure client secret
- * @return AZ_OK = Successfull initialization <br>
- * Other value = Initialization failed
+ * @retval An #az_result value indicating the result of the operation:
```
Comment (ahsonkhan) [inline sdk/core/core/inc/az_credentials.h:107]: For consistency, let's continue to use `@return`.
```suggestion
 * @return An #az_result value indicating the result of the operation:
```

See previous discussion on `@retval` vs `@return`.
https://github.com/Azure/azure-sdk-for-c/pull/622#discussion_r422412165

File: sdk/core/core/inc/az_credentials.h [hunk 5]
```diff
@@ -104,8 +104,9 @@ typedef struct
...
+ *         - #AZ_OK if successful
+ *         - Other value = Initialization failed
```
Comment (ahsonkhan) [inline sdk/core/core/inc/az_credentials.h:109]:
```suggestion
 *         - Other error code if initialization failed
```

File: sdk/core/core/inc/az_credentials.h [hunk 6]
```diff
@@ -104,8 +104,9 @@ typedef struct
  * @param tenant_id an Azure tenant ID
  * @param client_id an Azure client ID
  * @param client_secret an Azure client secret
- * @return AZ_OK = Successfull initialization <br>
- * Other value = Initialization failed
+ * @return An #az_result value indicating the result of the operation:
```
Comment (antkmsft) [inline sdk/core/core/inc/az_credentials.h:107]: I think we should be using `@retval` across all our docs. For this PR, it is fine (unless you want to solve this ultimately right now in this PR). But we should not forget this, let's discuss switching to `@retval` on our meetings.

Comment (ahsonkhan) [reply sdk/core/core/inc/az_credentials.h:107]: @antkmsft - please file an issue and let's address it across the board if there is a strong desire. I am fine with what we have currently, but also like the mixed approach:
https://github.com/Azure/azure-sdk-for-c/pull/622#discussion_r422412165

**Pull request merged**